### ISSUE TITLE
ts2532 issue in closure with strictNullChecks

### ISSUE THREAD

**Issue by thinkkevin** [microsoft/TypeScript#39156]
**TypeScript Version:**  3.9.2

**Search Terms:**

**Expected behavior:** No ts2532 error

**Actual behavior:** 

**Related Issues:**

**Code**
```ts
type Problem = {
  a: string
  b?: string;
}

const problem1: Problem = {
  a: "1",
  b: "2"
};

if (problem1.b) {
  const r = [1, 2, 3].map((num) => problem1.b.toUpperCase() + num);  // has issue ts2532
  console.log(r);
}

```

<details><summary><b>Output</b></summary>

```ts
"use strict";
const problem1 = {
    a: "1",
    b: "2"
};
if (problem1.b) {
    const r = [1, 2, 3].map((num) => problem1.b.toUpperCase() + num); // has issue ts2532
    console.log(r);
}

```

</details>

<details><summary><b>Compiler Options</b></summary>

```json
{
  "compilerOptions": {
    "noImplicitAny": true,
    "strictNullChecks": true,
    "strictFunctionTypes": true,
    "strictPropertyInitialization": true,
    "strictBindCallApply": true,
    "noImplicitThis": true,
    "noImplicitReturns": true,
    "useDefineForClassFields": false,
    "alwaysStrict": true,
    "allowUnreachableCode": false,
    "allowUnusedLabels": false,
    "downlevelIteration": false,
    "noEmitHelpers": false,
    "noLib": false,
    "noStrictGenericChecks": false,
    "noUnusedLocals": false,
    "noUnusedParameters": false,
    "esModuleInterop": true,
    "preserveConstEnums": false,
    "removeComments": false,
    "skipLibCheck": false,
    "checkJs": false,
    "allowJs": false,
    "declaration": true,
    "experimentalDecorators": false,
    "emitDecoratorMetadata": false,
    "target": "ES2017",
    "module": "ESNext"
  }
}
```

</details>

**Playground Link:** [Provided](https://www.typescriptlang.org/play/index.html?ssl=15&ssc=1&pln=1&pc=1#code/C4TwDgpgBACgTgewEYBsIFsoF4oG8BQUUAhgFxQDOwcAlgHYDmhUSA-OVbYwNz4C++fAGMEdKlDCJUGAIzl4yNJhwEiZKACIZGgDTMk5DQCYN-XvhoAzKAApJi2QDokASjzMRY4FDjYoAbRkdKCNggGYAXUd0YjAbGzoAV3Q3LAA+CSklGWdHYAQAVTBIOABhYgoIGzcAaigklO4iAHpmqAALCqgaCgpE6GAKIwBWMKMPUQoENEcUBAYbOBdeASA)

**Comment by DanielRosenwasser** [comment]
See https://github.com/microsoft/TypeScript/issues/9998
